src/cursor.c

```diff
@@ -882,6 +882,50 @@ cursor_button(struct wl_listener *listener, void *data)
 	}
 }
 
+bool
+handle_cursor_axis(struct server *server, struct cursor_context *ctx,
+		struct wlr_pointer_axis_event *event)
+{
+	struct mousebind *mousebind;
+	bool activated_any = false;
+	bool activated_any_frame = false;
+
+	uint32_t modifiers = wlr_keyboard_get_modifiers(
+			&server->seat.keyboard_group->keyboard);
+
+	uint32_t button = 0;
+	if (event->delta_discrete<0) {
+		button = BTN_GEAR_UP;
+	} else if (event->delta_discrete>0) {
+		button = BTN_GEAR_DOWN;
+	}
+
+	if (!button
+			|| event->source != WLR_AXIS_SOURCE_WHEEL
+			|| event->orientation != WLR_AXIS_ORIENTATION_VERTICAL) {
+		wlr_log(WLR_ERROR, "Failed to handle cursor axis event");
```

Comment:
Not sure we need the `WLR_ERROR` here, `WLR_DEBUG` might be enough (if the log is required at all).
In what cases can this go wrong?

Reply:
Right!
I guess one not supported case would be wlr_pointer_axis_event containing
delta rather than delta_discrete (continuous mouse wheel?)?
Changed log to WLR_DEBUG, but will gladly remove it completely, your call.

Reply:
AFAIR continuous mouse wheel scrolling is internally translated in wlroots into discrete steps of 120 (or the other way round), not completely sure right now.

I think the debug log is fine for now, we can always change it later when we figure out that it unnecessarily spams the log.